backend/src/main/java/gov/cdc/usds/simplereport/service/DiseaseCacheService.java

```diff
@@ -0,0 +1,29 @@
+package gov.cdc.usds.simplereport.service;
+
+import static gov.cdc.usds.simplereport.config.CachingConfig.SUPPORTED_DISEASE_ID_MAP;
+
+import gov.cdc.usds.simplereport.db.model.SupportedDisease;
+import gov.cdc.usds.simplereport.db.repository.SupportedDiseaseRepository;
+import java.util.Map;
+import java.util.UUID;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.stereotype.Service;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class DiseaseCacheService {
+
+  private final SupportedDiseaseRepository _supportedDiseaseRepo;
+
+  @Cacheable(SUPPORTED_DISEASE_ID_MAP)
+  public Map<UUID, SupportedDisease> getKnownSupportedDiseasesMap() {
+    log.info("generating " + SUPPORTED_DISEASE_ID_MAP + " cache");
```

Comment:
https://stackoverflow.com/questions/12115996/spring-cache-cacheable-method-ignored-when-called-from-within-the-same-class